From stable-bounces@linux.kernel.org  Fri Feb 24 13:09:14 2006
Date: Fri, 24 Feb 2006 13:04:23 -0800
From: akpm@osdl.org
To: torvalds@osdl.org
Cc: philipp.marek@bmlv.gv.at, stable@kernel.org
Subject: [PATCH] ramfs: update dir mtime and ctime


From: Andrew Morton <akpm@osdl.org>

Phil Marek <philipp.marek@bmlv.gv.at> points out that ramfs forgets to update
a directory's mtime and ctime when it is modified.

Cc: <stable@kernel.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 fs/ramfs/inode.c |    2 ++
 1 files changed, 2 insertions(+)

--- linux-2.6.15.4.orig/fs/ramfs/inode.c
+++ linux-2.6.15.4/fs/ramfs/inode.c
@@ -27,6 +27,7 @@
 #include <linux/fs.h>
 #include <linux/pagemap.h>
 #include <linux/highmem.h>
+#include <linux/time.h>
 #include <linux/init.h>
 #include <linux/string.h>
 #include <linux/smp_lock.h>
@@ -105,6 +106,7 @@ ramfs_mknod(struct inode *dir, struct de
 		d_instantiate(dentry, inode);
 		dget(dentry);	/* Extra count - pin the dentry in core */
 		error = 0;
+		dir->i_mtime = dir->i_ctime = CURRENT_TIME;
 	}
 	return error;
 }
